connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -218,130 +212,140 @@ object KustoWriter {
                                    parameters: KustoWriteResource): Seq[(CloudBlockBlob, Long)] = {
     import parameters._
 
-    var kustoClient = KustoClientCache.getClient(coordinates.cluster, authentication)
-    var initialStorageUri = kustoClient.getNewTempBlobReference
+    val kustoClient = KustoClientCache.getClient(coordinates.cluster, authentication)
+    val initialStorageUri = kustoClient.getNewTempBlobReference
     val initialContainer: CloudBlobContainer = new CloudBlobContainer(new URI(initialStorageUri))
+
     //This blobWriter will be used later to write the rows to blob storage from which it will be ingested to Kusto
     val initialBlobWriter: BlobWriteResource = createBlobWriter(schema, coordinates, tmpTableName, initialContainer)
     val dateFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", TimeZone.getTimeZone(writeOptions.timeZone))
 
     // Serialize rows to ingest and send to blob storage.
-    val (currentFileSize, currentBlobWriter, blobsToIngest) = rows.foldLeft[(Long, BlobWriteResource, Seq[(CloudBlockBlob, Long)])]((0, initialBlobWriter, Seq())) {
-      case ((size, blobWriter, blobsCreated), row) =>
-        val csvRowResult: CsvRowResult = convertRowToCSV(row, schema, dateFormat)
-        blobWriter.csvWriter.writeRow(csvRowResult.formattedRow)
-        val newTotalSize = size + csvRowResult.rowByteSize
+    val (currentBlobWriter, blobsToIngest) = rows.foldLeft[(BlobWriteResource, Seq[(CloudBlockBlob, Long)])]((initialBlobWriter, Seq())) {
+      case ((blobWriter, blobsCreated), row) =>
+
+        writeRowAsCSV(row, schema, dateFormat, blobWriter.csvWriter)
 
         // Check if we crossed the threshold for a single blob to ingest.
         // If so, add current blob to the list of blobs to ingest, and open a new blob writer
-        if (newTotalSize < maxBlobSize) {
-          (newTotalSize, blobWriter, blobsCreated)
+        if (blobWriter.csvWriter.getProgress < maxBlobSize) {
```

Comment:
this helps make the point I wrote above, a more readable version of this would be:
```scala
val shouldRoll = lobWriter.csvWriter.currentSize < maxBlobSize;
if (shouldRoll ) {
```

That way no comment is needed..